payload/src/main/java/io/jenkins/jenkinsfile/runner/Runner.java

```diff
@@ -113,4 +137,13 @@ private void writeLogTo(PrintStream out) throws IOException, InterruptedExceptio
             }
         }
     }
+
+    private void addCredentials(Credentials creds) throws IOException {
+        CredentialsStore store = getStore();
+        if (store == null) {
+            throw new RuntimeException("Credentials specified but could not find credentials store");
```

Comment:
Please use checked exceptions and propagate them correctly

Reply:
It's been too long - any suggestions as to what exception would be appropriate here?

Reply:
Decided to go with `CredentialsUnavailableException`.